tests/RobotFramework/tests/cumulocity/log/plugins/journald

```diff
@@ -0,0 +1,108 @@
+#!/bin/bash
+
+set -euo pipefail
+
+usage() {
+    echo "Usage: $0 <command> [args...]"
+    echo "Commands:"
+    echo "  list                                List all log types supported by this plugin"
+    echo "  get <type>                          Get logs for the specified type"
+    echo "      [--since <timestamp>]"
+    echo "      [--until <timestamp>]"
+    echo "      [--filter <text-filter>]"
+    echo "      [--tail <lines-from-end>]"
+    exit 1
+}
+
+list_log_types() {
+    echo ":::begin-tedge:::"
+    systemctl list-units --type=service --no-legend --no-pager --plain \
+        | awk '{ sub(/\.service$/, "", $1); print $1 }'
+    echo ":::end-tedge:::"
+}
+
+get_logs() {
+    local log_type="$1"
+    shift
+    
+    local since_arg=""
+    local until_arg=""
+    local filter_arg=""
+    local tail_arg=""
+    
+    while [[ $# -gt 0 ]]; do
+        case $1 in
+            --since)
+                since_arg="--since=@$2"
+                shift 2
+                ;;
+            --until)
+                until_arg="--until=@$2"
+                shift 2
+                ;;
+            --filter)
+                filter_arg="$2"
+                shift 2
+                ;;
+            --tail)
+                tail_arg="--lines=$2"
+                shift 2
+                ;;
+            *)
+                echo "Unknown option: $1" >&2
+                exit 1
+                ;;
+        esac
+    done
+    
+    local cmd="journalctl --no-pager --output=short-iso --unit=${log_type}"
+    
+    if [[ -n "$since_arg" ]]; then
+        cmd="$cmd $since_arg"
+    fi
+    
+    if [[ -n "$until_arg" ]]; then
+        cmd="$cmd $until_arg"
+    fi
+    
+    if [[ -n "$tail_arg" ]]; then
+        cmd="$cmd $tail_arg"
+    fi
+    
+    # Execute journalctl and apply text filter if specified
+    echo ":::begin-tedge:::"
+    if [[ -n "$filter_arg" ]]; then
+        eval "$cmd" | grep -i "$filter_arg"
+    else
+        eval "$cmd"
+    fi
+    echo ":::end-tedge:::"
```

Comment:
Do these `:::begin-tedge:::` and `:::end-tedge:::` header and trailer actually help? I feel the contrary, it makes things more fragile.

For instance the current version of this plugin never emits the trailer when there is no matching rows (because of the early exit on non 0 exit code - `set -e`). This can be fixed:


```suggestion
    echo ":::begin-tedge:::"
    if [[ -n "$filter_arg" ]]; then
        eval "$cmd" | (grep -i "$filter_arg" || true)
    else
        eval "$cmd"
    fi
    echo ":::end-tedge:::"
```

However, this puts more burden to the script writer than a proper handling of the error stream.

Reply:
For this type of interface, we can stick to plain stdout as we're generally only running one command to actually generate the logs, and those commands are already using stdout as the output of the logs...It is always useful to see how close we can get to using the plain `journalctl` command without even writing a plugin (though I know it won't work, but good to align closer to that rather than building too much of an abstraction).

Reply:
Resolved by a17e59f